esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -6255,16 +6261,25 @@ bool CWsDfuEx::onDFUFileCreate(IEspContext &context, IEspDFUFileCreateRequest &r
         }
         context.ensureFeatureAccess(FEATURE_URL, accessType, ECLWATCH_DFU_ACCESS_DENIED, "WsDfu::DFUFileCreate: Permission denied.");
 
-        StringBuffer fileName = req.getName();
+        const char *fileName = requests.getName();
         const char *clusterName = requests.getCluster();
         const char *recordDefinition = req.getECLRecordDefinition();
-        if (fileName.isEmpty())
+        if (isEmptyString(fileName))
              throw MakeStringException(ECLWATCH_INVALID_INPUT, "No Name defined.");
         if (isEmptyString(clusterName))
              throw MakeStringException(ECLWATCH_INVALID_INPUT, "No Cluster defined.");
         if (isEmptyString(recordDefinition))
              throw MakeStringException(ECLWATCH_INVALID_INPUT, "No ECLRecordDefinition defined.");
 
+        StringBuffer groupName;
+        if (!createWsDFUCreateGroupName(req.getPartLocations(), groupName))
+            throw MakeStringException(ECLWATCH_INVALID_INPUT, "Failed in createWsDFUCreateGroupName().");
+
+        Owned<IConstWUClusterInfo> clusterInfo = getTargetClusterInfo(clusterName);
+        ClusterType clusterType = clusterInfo->getPlatform();
```

Comment:
We need a nullptr check on clusterInfo. getTargetClusterInfo returns a nullptr on invalid clusterName

Reply:
I will add the check.